src/snmp_standard/mode/interfaces.pm

```diff
@@ -1207,9 +1207,10 @@ sub get_selection {
     my $all_ids = $self->{statefile_cache}->get(name => 'all_ids');
     if (!defined($self->{option_results}->{use_name}) && defined($self->{option_results}->{interface}) 
         && $self->{no_interfaceid_options} == 0) {
-        foreach (@{$all_ids}) {
-            if ($self->{option_results}->{interface} =~ /(^|\s|,)$_(\s*,|$)/) {
-                $self->add_selected_interface(id => $_);
+        my $regex = $self->{option_results}->{interface};
+        for my $id (@{$all_ids}) {
+            if ($id =~ /(^|\s|,)$regex(\s*,|$)/) {
```

Comment:
This changes the behaviour a lot. We should add an option to switch between comma-separated list of ids and a regexp.